src/components/PreviewMedia/index.tsx

```diff
@@ -0,0 +1,104 @@
+'use client';
+import 'swiper/css';
+import 'swiper/css/keyboard';
+import 'swiper/css/navigation';
+
+import { ArrowLeftIcon, ArrowRightIcon } from '@heroicons/react/24/outline';
+import { useMemo, useRef } from 'react';
+import { Keyboard, Navigation } from 'swiper/modules';
+import { Swiper, SwiperSlide } from 'swiper/react';
+
+import { PostActionsWithGrid } from '@/components/Actions/index.js';
+import { ClickableButton } from '@/components/ClickableButton.js';
+import { CloseButton } from '@/components/CloseButton.js';
+import { Modal } from '@/components/Modal.js';
+import { PreviewContent } from '@/components/PreviewMedia/PreviewContent.js';
+import type { Source } from '@/constants/enum.js';
+import { EMPTY_LIST, SUPPORTED_PREVIEW_MEDIA_TYPES } from '@/constants/index.js';
+import { useIsMedium } from '@/hooks/useMediaQuery.js';
+import type { Attachment, Post } from '@/providers/types/SocialMedia.js';
+
+interface PreviewMediaProps {
+    index: string;
+    open: boolean;
+    source?: Source;
+    post?: Post;
+    medias?: Attachment[];
+    showAction?: boolean;
+    onClose: () => void;
+}
+
+export function PreviewMedia({ post, source, medias, index, open, showAction = true, onClose }: PreviewMediaProps) {
+    const prevRef = useRef<HTMLButtonElement>(null);
+    const nextRef = useRef<HTMLButtonElement>(null);
+    const isMedium = useIsMedium();
+
+    const images = useMemo(() => {
+        if (medias) return [...medias];
```

Comment:
You won't mutate it, right? No need to create a copy.
```suggestion
        if (medias) return medias;
```